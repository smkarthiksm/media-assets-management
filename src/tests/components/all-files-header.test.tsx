import '@testing-library/jest-dom';
import AllFilesHeaderComponent from '../../components/all-files-header/all-files-header';
import { renderWithProviders } from '../test-utility';
import { fireEvent, screen } from '@testing-library/react';
import { allFilesStateStub } from '../stubs/all-files-state.stub';

describe('AllFilesHeaderComponent', () => {
  test('should render', () => {
    const { container } = renderWithProviders(<AllFilesHeaderComponent />);
    expect(container).toMatchSnapshot();
  });

  test('should update search text field when input changes', () => {
    renderWithProviders(<AllFilesHeaderComponent />);
    const searchField = screen.getByPlaceholderText(
      'Search by title, album, artist',
    ) as HTMLInputElement;
    fireEvent.change(searchField, { target: { value: 'Test' } });
    expect(searchField.value).toEqual('Test');
  });

  test('should reset files with the cached files when search text field becomes empty', () => {
    const { store } = renderWithProviders(<AllFilesHeaderComponent />, {
      preloadedState: {
        allFiles: {
          allFiles: [allFilesStateStub.allFiles[0]],
          cachedAllFiles: allFilesStateStub.allFiles,
          searchValue: 'test',
        },
      },
    });
    const searchField = screen.getByPlaceholderText(
      'Search by title, album, artist',
    ) as HTMLInputElement;

    expect(store.getState().allFiles.allFiles).toEqual([
      allFilesStateStub.allFiles[0],
    ]);
    fireEvent.change(searchField, { target: { value: '' } });

    expect(store.getState().allFiles.allFiles).toEqual(
      allFilesStateStub.allFiles,
    );
  });

  // added for code coverage
  test('should return file when title or album or artist matches', () => {
    const { store } = renderWithProviders(<AllFilesHeaderComponent />, {
      preloadedState: {
        allFiles: {
          allFiles: allFilesStateStub.allFiles,
          cachedAllFiles: allFilesStateStub.cachedAllFiles,
          searchValue: '',
        },
      },
    });
    const searchField = screen.getByPlaceholderText(
      'Search by title, album, artist',
    ) as HTMLInputElement;
    fireEvent.change(searchField, { target: { value: 'testAlbum1' } });
    expect(store.getState().allFiles.allFiles).toEqual([
      allFilesStateStub.allFiles[0],
    ]);
    fireEvent.change(searchField, { target: { value: 'testArtist1' } });
    expect(store.getState().allFiles.allFiles).toEqual([
      allFilesStateStub.allFiles[0],
    ]);
    fireEvent.change(searchField, { target: { value: 'testTitle1' } });
    expect(store.getState().allFiles.allFiles).toEqual([
      allFilesStateStub.allFiles[0],
    ]);
  });

  test('should reset and open modal to upload files', () => {
    const { store } = renderWithProviders(<AllFilesHeaderComponent />);
    const uploadButton = screen.getByRole('button', {
      name: 'Upload',
    });
    expect(
      store.getState().uploadFileModal.isUploadFileModalVisible,
    ).toBeFalsy();
    fireEvent.click(uploadButton);
    expect(
      store.getState().uploadFileModal.isUploadFileModalVisible,
    ).toBeTruthy();
  });
});
