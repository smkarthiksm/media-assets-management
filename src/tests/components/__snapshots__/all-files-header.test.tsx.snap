// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllFilesHeaderComponent should render 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 css-spompt-MuiGrid-root"
  >
    <div
      class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
    >
      Manage your files
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-container css-1xzqqmo-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
    >
      <div
        class="MuiFormControl-root MuiTextField-root input-field css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            id=":r0:"
            placeholder="Search by title, album, artist"
            type="text"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-ihdtdm"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-ym3k52-MuiGrid-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Upload
        <span
          class="MuiButton-icon MuiButton-endIcon MuiButton-iconSizeMedium css-9tj150-MuiButton-endIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AddIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
